timemachine/cpp/src/nonbonded.cu

```diff
@@ -303,6 +303,23 @@ void Nonbonded<RealType, Interpolated>::execute_device(
             k_arange<<<B, tpb, 0, stream>>>(N, d_perm_);
             gpuErrchk(cudaPeekAtLastError());
         }
+
+        // Verify that the cutoff and box size are valid together. If cutoff is greater than half the box
+        // then a particle can interact with multiple periodic copies.
+        std::vector<double> h_box(9);
+        gpuErrchk(cudaMemcpyAsync(&h_box[0], d_box, 3*3*sizeof(*d_box), cudaMemcpyDeviceToHost, stream));
+        const double twice_sq_cutoff = 2 * ((cutoff_+nblist_padding_) * (cutoff_+nblist_padding_));
+
+        cudaStreamSynchronize(stream);
+        // Use l2 norm on each row to determine if cutoff is too small
+        for (int i = 0; i < 3; i++) {
+            double row_norm_sq = h_box[i*3+0]*h_box[i*3+0] + h_box[i*3+1]*h_box[i*3+1] + h_box[i*3+2]*h_box[i*3+2];
```

Comment:
looks right to me, note that we should document somewhere that the box 3x3 matrix corresponds to:
```
[
   [u_x,u_y,u_z],
   [v_x,v_y,v_z], 
   [w_x,w_y,w_z],
]
```

where `u,v,w` correspond to be the box vectors. Different packages have different conventions.